src/core/vote-core.js

```diff
@@ -23,30 +26,48 @@ function createOrUpdateVote(opts) {
     UPDATE feed_items
     SET hot_score = ?
     WHERE id = ?
+    RETURNING
+      (SELECT
+        users.team_id
+      FROM
+        feed_items
+        LEFT JOIN users ON users.id = feed_items.user_id
+      WHERE feed_items.id = ?)
   `;
 
-  const updateVoteParams = [opts.value, opts.client.id, opts.feedItemId,
-    opts.value, opts.client.id, opts.feedItemId];
+  const updateVoteParams = [
+    opts.value, opts.ip, opts.client.id, opts.feedItemId,
+    opts.value, opts.ip, opts.client.id, opts.feedItemId,
+  ];
 
-  return knex.transaction(function(trx) {
-    return trx.raw(updateVoteSql, updateVoteParams)
-      .then(result => trx.raw(selectVotesSql, [opts.feedItemId]))
-      .then(result =>
-        trx.raw(updateFeedItemSql, [
-            hotScore(
-              result.rows[0]['votes'],
-              result.rows[0]['age']
-            ),
-            opts.feedItemId,
-          ])
-      )
-      .then(result => undefined)
-      .catch(err => {
-        let error = new Error('No such feed item id: ' + opts.feedItemId);
-        error.status = 404;
-        throw error;
-      });
-  });
+  return opts.client.isBanned
+    ? Promise.resolve()
+    : knex.transaction((trx) => trx.raw(updateVoteSql, updateVoteParams)
+        .then(result => trx.raw(selectVotesSql, [opts.feedItemId]))
+        .then(result =>
+          trx.raw(updateFeedItemSql, [
+              hotScore(
+                result.rows[0]['votes'],
+                result.rows[0]['age']
+              ),
+              opts.feedItemId,
+              opts.feedItemId,
+            ])
+        )
+        .then(result => biasCore.calculateBias({
```

Comment:
Is this a heavy operation? I.e. does it affect a lot to the performance of `/vote` endpoint?